src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -541,12 +593,18 @@ def _bracket_sensitive_look_ahead_match(cls, segments, matchers, parse_context):
                 # Now check have we closed all our brackets?
                 if bracket_stack:
                     # No we haven't.
-                    # TODO: Format this better
-                    raise SQLParseError(
-                        "Couldn't find closing bracket for opening bracket.",
-                        segment=bracket_stack.pop(),
+                    # Check that the unclosed brackets are definite
+                    definite_bracket_stack = list(
+                        filter(lambda b: b[1], zip(bracket_stack, definite_stack))
                     )
 
+                    # TODO: Format this better
+                    if definite_bracket_stack:

```

Comment:
Here it's the same, I'm adding a conditional to relax the closing bracket checks in case the brackets are marked as not definite